controller/konnect/reconciler_kongplugin.go

```diff
@@ -0,0 +1,278 @@
+package konnect
+
+import (
+	"context"
+	"strings"
+
+	"github.com/samber/lo"
+	k8serrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+
+	"github.com/kong/gateway-operator/controller/pkg/log"
+	"github.com/kong/gateway-operator/pkg/consts"
+	k8sutils "github.com/kong/gateway-operator/pkg/utils/kubernetes"
+
+	configurationv1 "github.com/kong/kubernetes-configuration/api/configuration/v1"
+	configurationv1alpha1 "github.com/kong/kubernetes-configuration/api/configuration/v1alpha1"
+)
+
+// KongPluginReconciler reconciles a KongPlugin object.
+type KongPluginReconciler struct {
+	developmentMode bool
+	client          client.Client
+}
+
+// NewKongPluginReconciler creates a new KongPluginReconciler.
+func NewKongPluginReconciler(
+	developmentMode bool,
+	client client.Client,
+) *KongPluginReconciler {
+	return &KongPluginReconciler{
+		developmentMode: developmentMode,
+		client:          client,
+	}
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *KongPluginReconciler) SetupWithManager(_ context.Context, mgr ctrl.Manager) error {
+	return ctrl.NewControllerManagedBy(mgr).
+		Named("KongPlugin").
+		For(&configurationv1.KongPlugin{}).
+		Watches(
+			&configurationv1alpha1.KongPluginBinding{},
+			handler.EnqueueRequestsFromMapFunc(r.mapKongPluginBindings),
+		).
+		Watches(
+			&configurationv1alpha1.KongService{},
+			handler.EnqueueRequestsFromMapFunc(r.mapKongServices),
+		).
+		Complete(r)
+}
+
+// Reconcile reconciles a KongPlugin object.
+func (r *KongPluginReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	var (
+		entityTypeName = "KongPlugin"
+		logger         = log.GetLogger(ctx, entityTypeName, r.developmentMode)
+	)
+
+	// Fetch the KongPlugin instance
+	var kongPlugin configurationv1.KongPlugin
+	if err := r.client.Get(ctx, req.NamespacedName, &kongPlugin); err != nil {
+		if k8serrors.IsNotFound(err) {
+			return ctrl.Result{}, nil
+		}
+		return ctrl.Result{}, err
+	}
+
+	log.Debug(logger, "reconciling", kongPlugin)
+
+	// Get the pluginBindings that use this KongPlugin
+	pluginBindings := []configurationv1alpha1.KongPluginBinding{}
+	referencingBindingList := configurationv1alpha1.KongPluginBindingList{}
+	err := r.client.List(
+		ctx,
+		&referencingBindingList,
+		client.MatchingFields{
+			IndexFieldKongPluginBindingKongPluginReference: kongPlugin.Name,
```

Comment:
According to the `ExtractValue` func in here: https://github.com/Kong/gateway-operator/blob/4731ea64685befe828bbdcdc648d6f5e11660271/controller/konnect/index_kongpluginbinding.go#L60-L70

it extract the value from the plugin binding objects as `<ns>/<name>`. Doesn't that mean that this won't match and we need to prepend the `kongPlugin.Name` with `kongPlugin.Namespace + "/"`?

Reply:
yep, the integration test was broken indeed :). I added this indexing in a rush and I overlooked this aspect. Fixed in 285eb9c353c58691f0a4477a73ed27eefa694738